test/jprob_symbol_indexing.jl

```diff
@@ -6,14 +6,16 @@ affect1!(integ) = (integ.u[1] += 1)
 affect2!(integ) = (integ.u[2] += 1)
 crj1 = ConstantRateJump(rate1, affect1!)
 crj2 = ConstantRateJump(rate2, affect2!)
-g = DiscreteFunction((du, u, p, t) -> nothing; syms = [:a, :b], paramsyms = [:p1, :p2])
+g = DiscreteFunction((du, u, p, t) -> nothing; sys = SymbolicIndexingInterface.SymbolCache([:a, :b], [:p1, :p2], :t))
 dprob = DiscreteProblem(g, [0, 10], (0.0, 10.0), [1.0, 2.0])
 jprob = JumpProblem(dprob, Direct(), crj1, crj2)
 
 # runs the tests
 @test jprob[:a] == 0
 @test jprob[:b] == 10
-@test jprob[:p1] == 1.0
-@test jprob[:p2] == 2.0
 
-# tests for setindex (e.g. `jprob[:a] = 10`) not possible, this requires the problem to have a .f.sys filed.,
+# these are no longer supported by SciMLBase
+@test getp(jprob,:p1)(jprob) == 1.0
+@test getp(jprob,:p2)(jprob) == 2.0
+@test jprob.ps[:p1] == 1.0
+@test jprob.ps[:p2] == 2.0
```

Comment:
Where has the dispatch been added to ensure `update_parameters!(prob.massaction_jump, prob.prob.p)` is called when these parameter changes are made?